src/WebJobs.Script.WebHost/SecretManager.cs

```diff
@@ -81,14 +89,29 @@ public virtual FunctionSecrets GetFunctionSecrets(string functionName)
                 }
                 else
                 {
-                    // initialize with empty instance
-                    secrets = new FunctionSecrets();
+                    // initialize with new secrets and save it
```

Comment:
How is this going to work? We only get (and therefore generate) this when the function is invoked it seems. However, the value is needed to invoke the function in the first place. Also, how will the portal show the value in the URL box before it is generated?